iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/tsfile/timeindex/FileTimeIndexCacheRecorder.java

```diff
@@ -0,0 +1,241 @@
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import static org.apache.iotdb.commons.utils.FileUtils.deleteDirectoryAndEmptyParent;
+
+public class FileTimeIndexCacheRecorder {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(FileTimeIndexCacheRecorder.class);
+
+  private static final int VERSION = 0;
+
+  protected static final String FILE_NAME = "FileTimeIndexCache_" + VERSION;
+
+  private final ScheduledExecutorService recordFileIndexThread;
+
+  private final BlockingQueue<Runnable> taskQueue = new LinkedBlockingQueue<>();
+
+  private final Map<Integer, Map<Long, FileTimeIndexCacheWriter>> writerMap =
+      new ConcurrentHashMap<>();
+
+  private FileTimeIndexCacheRecorder() {
+    recordFileIndexThread =
+        IoTDBThreadPoolFactory.newSingleThreadScheduledExecutor(
+            ThreadName.FILE_TIMEINDEX_RECORD.getName());
+    ScheduledExecutorUtil.safelyScheduleWithFixedDelay(
+        recordFileIndexThread, this::executeTasks, 10, 10, TimeUnit.MILLISECONDS);
+  }
+
+  private void executeTasks() {
+    Runnable task;
+    while ((task = taskQueue.poll()) != null) {
+      recordFileIndexThread.submit(task);
+    }
+  }
+
+  public void logFileTimeIndexes(List<TsFileResource> tsFileResources) {
+    if (!tsFileResources.isEmpty()) {
+      TsFileResource firstResource = tsFileResources.get(0);
+      TsFileID tsFileID = firstResource.getTsFileID();
+      int dataRegionId = tsFileID.regionId;
+      long partitionId = tsFileID.timePartitionId;
+      File dataRegionSysDir =
+          StorageEngine.getDataRegionSystemDir(
+              firstResource.getDatabaseName(), firstResource.getDataRegionId());
+      FileTimeIndexCacheWriter writer = getWriter(dataRegionId, partitionId, dataRegionSysDir);
+      boolean result =
+          taskQueue.offer(
+              () -> {
+                try {
+                  ByteBuffer buffer = ByteBuffer.allocate(5 * Long.BYTES * tsFileResources.size());
+                  for (TsFileResource tsFileResource : tsFileResources) {
+                    tsFileResource.serializeFileTimeIndexToByteBuffer(buffer);
+                  }
+                  buffer.flip();
+                  writer.write(buffer);
+                } catch (IOException e) {
+                  LOGGER.warn("Meet error when record FileTimeIndexCache: {}", e.getMessage());
+                }
+              });
+      if (!result) {
+        LOGGER.warn("Meet error when record FileTimeIndexCache");
+      }
+    }
+  }
+
+  public void logFileTimeIndex(TsFileResource tsFileResource) {
```

Comment:
The two methods are highly similar. How about using `public void logFileTimeIndex(TsFileResource... tsFileResources)`

Reply:
makes sense